### PR TITLE
Add Azure integration

### PR DESCRIPTION
**Pull request by dabcoder**
See https://github.com/terraform-providers/terraform-provider-datadog/issues/196 for context.

### CODE REVIEW

File: integrations.go [hunk 1]
```diff
@@ -274,3 +274,68 @@ func (client *Client) UpdateIntegrationGCP(cir *IntegrationGCPUpdateRequest) err
...
+// IntegrationAzureCreateRequest defines the request payload for creating Datadog-Azure CloudPlatform integration.
+type IntegrationAzureCreateRequest struct {
+	TenantName   *string `json:"tenant_name"`
+	ClientID     *string `json:"client_id"`
+	ClientSecret *string `json:"client_secret"`
+	HostFilters  *string `json:"host_filters,omitempty"`
+}
+
+// IntegrationAzureUpdateHostFiltersRequest defines the request payload for creating Datadog-Azure CloudPlatform integration.
+type IntegrationAzureUpdateHostFiltersRequest struct {
```
Comment (bkabrda) [inline integrations.go]: The API documentation says that `client_secret` is required for this request as well. I'd prefer making all the structures from this PR one - `IntegrationAzure` - perhaps with the exception of `IntegrationAzureUpdateAccountRequest` which has some unique arguments (`new_client_id` and `new_client_name` - BTW I couldn't find any documentation of these in the API docs, where are they defined?).

File: integrations.go [hunk 2]
```diff
@@ -274,3 +274,68 @@ func (client *Client) UpdateIntegrationGCP(cir *IntegrationGCPUpdateRequest) err
...
+	TenantName  *string `json:"tenant_name"`
+	ClientID    *string `json:"client_id"`
+	HostFilters *string `json:"host_filters"`
+}
+
+// IntegrationAzureUpdateAccountRequest defines the request payload for creating Datadog-Azure CloudPlatform integration.
```
Comment (bkabrda) [inline integrations.go]: This comment seems to be wrong, this is definitely not a create payload.

File: integrations.go (hunk 1 above)
Comment (dabcoder) [reply integrations.go]: Yes, so this is what we discussed with @Ricky-Thomas, no documentation for this endpoint just yet. 
I can however work on the `struct`s, thanks for the review.

**Comment by dabcoder** [comment]
Still WIP as this is pending feedback for the `IntegrationAzureUpdateAccountRequest` / Update Azure account.

File: integrations.go (hunk 1 above)
Comment (Ricky-Thomas) [reply integrations.go]: There are docs for that endpoint available now here: https://docs.datadoghq.com/api/?lang=bash#integration-azure

**Comment by Ricky-Thomas** [comment]
The docs are currently up to date with all the Azure functionality that should be added here. Lets make sure all the existing functionality is accounted for here.

**Comment by gzussa** [comment]
We (Datadog) implemented this feature in our Official Go Client https://github.com/DataDog/datadog-api-client-go. Closing this PR. If the open source community want to add this feature into this client. Feel free to reopen this PR.

**Pull request closed without merging**